contracts/artifacts/DelegatedOptIn.approval.teal

```diff
@@ -0,0 +1,530 @@
+#pragma version 8
+	b main
+
+getSenderReceiverHash:
+	proto 2 1
+
+	// ./contracts/delegated_optin_app.algo.ts:36
+	// return sha256(concat(sender, receiverOrSigner));
+	frame_dig -1 // sender: address
+	frame_dig -2 // receiverOrSigner: address
+	concat
+	sha256
+	retsub
+
+bare_route_create:
+	txn OnCompletion
+	int NoOp
+	==
+	txn ApplicationID
+	int 0
+	==
+	&&
+	assert
+
+	// no dupn needed
+	callsub create
+	int 1
+	return
+
+create:
+	proto 0 0
+
+	// ./contracts/delegated_optin_app.algo.ts:41
+	// this.assetMBR.set(100_000)
+	byte "assetMBR"
+	int 100_000
+	app_global_put
+	retsub
+
+abi_route_setSigVerificationAddress:
+	txn OnCompletion
+	int NoOp
+	==
+	txn ApplicationID
+	int 0
+	!=
+	&&
+	assert
+
+	// no dupn needed
+	txna ApplicationArgs 1
+	callsub setSigVerificationAddress
+	int 1
+	return
+
+setSigVerificationAddress:
+	proto 1 0
+
+	// ./contracts/delegated_optin_app.algo.ts:53
+	// assert(this.txn.sender === this.app.creator)
+	txn Sender
+	txna Applications 0
+	app_params_get AppCreator
+	assert
+	==
+	assert
+
+	// ./contracts/delegated_optin_app.algo.ts:54
+	// assert(!this.sigVerificationAddress.exists())
+	txna Applications 0
+	byte "sigVerificationAddress"
+	app_global_get_ex
+	swap
+	pop
+	!
+	assert
+
+	// ./contracts/delegated_optin_app.algo.ts:55
+	// this.sigVerificationAddress.set(lsig)
+	byte "sigVerificationAddress"
+	frame_dig -1 // lsig: address
+	app_global_put
+	retsub
```

Comment:
Unfortunately not because the verify program requires the programs of the opt-in lsigs, which require the app ID of this application. Although I suppose the deployment process could be to deploy an `int 1` approval program and then update the program after the fact with the hardcoded address.

Reply:
What "runtime" are you referring to? The lsig needs to have the app ID hardcoded otherwise the user will let ANY app approve opt-ins, which is not desirable.

Reply:
Right now, `verifier_lsig` is confirming that it's looking at a signature against `TMPL_OPEN_OPTIN_DATA` which I guess is the lsig bytecode, right? You could replace some bytes in `TMPL_OPEN_OPTIN_DATA` with the app id before calling `ed25519verify_bare`

Actually, I'm totally confused, it seems.  What the heck _is_ `TMPL_OPEN_OPTIN_DATA` and `TMPL_ADDRESS_OPTIN_DATA`. I thought it was bytecode, but I no longer think it is.

It's possible I have seriously misunderstood how this thing works.

Reply:
No you're right. It's the data that's signed when a user signs an lsig, which is the lsig tag ("Program") + lsig bytecode. 
If the verifier program needs to know the app ID, then we're back to square one where the `verifier_lsig` program depends on the app ID thus the `verifier_lsig` needs to be created after the app.